python/mlc_chat/compiler/quantization/group_quantization.py

```diff
@@ -102,24 +153,48 @@ def _dequantize(
         return dequantized_weight
 
     def quantize_weight(self, weight: NDArray) -> List[NDArray]:
-        """Quantize weight with group quantization"""
+        """
+        Quantize weight with group quantization
+
+        Parameters
+        ----------
+        weight : NDArray
+            The original weight.
+
+        Returns
+        ------
+        ret: List[NDArray]
+            The list of group quantized weights.
+        """
         assert weight.dtype == self.model_dtype
         assert len(weight.shape) == 2
-        bb = relax.BlockBuilder()
+        dev = weight.device
+        device_type = dev.MASK2STR[dev.device_type]
+        key = str((int(weight.shape[0]), int(weight.shape[1]), weight.dtype, device_type))
+        if key in self.prebuilt_quantize_func:
+            return self.prebuilt_quantize_func[key](weight)
+        bb = relax.BlockBuilder()  # pylint: disable=invalid-name
         weight_var = relax.Var("weight", relax.TensorStructInfo(weight.shape, self.model_dtype))
         with bb.function(name="quantize", params=[weight_var]):
             with bb.dataflow():
-                lv = bb.emit_te(self._quantize, weight_var)
-                gv = bb.emit_output(lv)
+                lv = bb.emit_te(self._quantize, weight_var)  # pylint: disable=invalid-name
+                gv = bb.emit_output(lv)  # pylint: disable=invalid-name
             bb.emit_func_output(gv)
         mod = bb.get()
-        with Target("cuda"):
-            mod = dl.ApplyDefaultSchedule(  # pylint: disable=not-callable
-                dl.gpu.Reduction(), dl.gpu.GeneralReduction(), dl.gpu.Fallback()
-            )(mod)
-        ex = relax.build(mod, "cuda")
-        dev = device("cuda", 0)
-        vm = relax.VirtualMachine(ex, dev)
+        if device_type in ["cuda", "rocm", "metal", "vulkan"]:
+            target = Target.from_device(dev)
```

Comment:
do not detect `target` if it's already defined from outside. Using `target = Target.current()` to check if there's any:

```python
with target:
   call-quantizer()
```

from outside

Reply:
fixed